acond/src/report.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::utils;
+use anyhow::{anyhow, Result};
+use nix::{ioctl_read, ioctl_readwrite};
+use openssl::sha;
+use std::{
+    convert::TryInto,
+    fs::File,
+    marker::PhantomData,
+    mem::{self, MaybeUninit},
+    os::unix::io::AsRawFd,
+};
+
+pub const EXTEND_RTMR_DATA_SIZE: usize = 0x30;
+pub const REPORT_DATA_SIZE: usize = 0x40;
+pub const REPORT_SIZE: usize = 0x400;
+pub const NUM_RTMRS: i32 = 4;
+
+const ATTEST_DEV_PATH: &str = "/dev/tdx_guest";
+const HEADER_SIZE: usize = 4;
+const REQ_BUF_SIZE: usize = 0x4000;
+const GET_QUOTE_IN_FLIGHT: u64 = 0xffffffffffffffff;
+const GET_QUOTE_SERVICE_UNAVAILABLE: u64 = 0x8000000000000001;
+
+#[repr(C)]
+pub struct __IncompleteArrayField<T>(PhantomData<T>, [T; 0]);
+
+#[repr(C)]
+pub struct TdxReportReq {
+    report_data: [u8; REPORT_DATA_SIZE],
+    td_report: [u8; REPORT_SIZE],
+}
+
+#[repr(C)]
+pub struct TdxExtendRtmrReq {
+    data: [u8; EXTEND_RTMR_DATA_SIZE],
+    index: u8,
+}
+
+#[repr(C)]
+struct TdxQuoteHdr {
+    version: u64,
+    status: u64,
+    in_len: u32,
+    out_len: u32,
+    data: __IncompleteArrayField<u64>,
+}
+
+#[repr(C)]
+pub struct TdxQuoteReq {
+    buf: u64,
+    len: u64,
+}
+
+ioctl_readwrite!(get_tdx_report, b'T', 0x01, TdxReportReq);
+ioctl_read!(extend_tdx_rtmr, b'T', 0x03, TdxExtendRtmrReq);
```

Comment:
OK. I will do the modification according to the latest kernel.
I didn't learn about this library ioctl_sys. Since nix is more common and is already used in this project, I just reuse its related modules.